src/Altinn.App.Core/Internal/Validation/ValidationService.cs

```diff
@@ -74,29 +79,31 @@ private Task<List<ValidationIssue>[]> RunTaskValidators(Instance instance, strin
         var taskValidators = _validatorFactory.GetTaskValidators(taskId);
 
         return Task.WhenAll(
-            taskValidators.Select(async tv =>
+            taskValidators.Select(async v =>
             {
+                using var activity = _telemetry?.StartRunTaskValidatorActivity(v);
                 try
                 {
                     _logger.LogDebug(
                         "Start running validator {validatorName} on task {taskId} in instance {instanceId}",
-                        tv.GetType().Name,
+                        v.GetType().Name,
                         taskId,
                         instance.Id
                     );
-                    var issues = await tv.ValidateTask(instance, taskId, language);
-                    issues.ForEach(i => i.Source = tv.ValidationSource); // Ensure that the source is set to the validator source
+                    var issues = await v.ValidateTask(instance, taskId, language);
+                    issues.ForEach(i => i.Source = v.ValidationSource); // Ensure that the source is set to the validator source
                     return issues;
                 }
                 catch (Exception e)
                 {
                     _logger.LogError(
                         e,
                         "Error while running validator {validatorName} on task {taskId} in instance {instanceId}",
-                        tv.GetType().Name,
+                        v.GetType().Name,
                         taskId,
                         instance.Id
                     );
+                    activity?.Errored(e);
```

Comment:
Need some docs on 
* When do we `RecordException`
* Instrumentation practice when calling into "user code"